Dockerfile

```diff
@@ -0,0 +1,52 @@
+# This dockerfile is optimized to have efficient layers caching.
+# for example composer install is ran only if composer related files are updated.
+# also modifying source, would not need to rebuild extensions layer.
+# Author: Elan Ruusamäe <glen@pld-linux.org>
+
+FROM php:7.3-fpm-alpine AS base
+
+RUN set -x \
+    && apk add --update --no-cache --virtual .build-deps ${PHPIZE_DEPS} \
+	&& pecl install mongodb && docker-php-ext-enable mongodb \
+    && apk del .build-deps
+
+# prepare sources
+FROM scratch AS source
+WORKDIR /app
+COPY . .
+# mkdir "vendor" dir, so the next stage can use external vendor optionally
+WORKDIR /app/vendor
+
+# install composer vendor
+FROM base AS build
+WORKDIR /app
+ARG COMPOSER_FLAGS="--no-interaction --no-suggest --ansi --no-dev"
+COPY --from=composer:1.8 /usr/bin/composer /usr/bin/
+
+COPY --from=source /app/composer.* ./
+COPY --from=source /app/vendor ./vendor
+
+# install in two steps to cache composer run based on composer.* files
+RUN composer require --update-no-dev --no-scripts alcaeus/mongo-php-adapter ^1.1
+RUN composer install $COMPOSER_FLAGS --no-scripts --no-autoloader
+
+# copy rest of the project. copy in order that is least to most changed
+COPY --from=source /app/external ./external
```

Comment:
If xhgui is running in isolation external isn't really required, as folks would be using the xhgui-collector package.

Reply:
indeed. removed